index.js

```diff
@@ -130,122 +134,120 @@ function serializeResponse (requestId, protocolData) {
   const writer = new Writer()
   writeProtocolData(writer, protocolData)
 
-  return writeEnvelope(TYPE_RESPONSE, requestId, writer.getBuffer())
+  return writeCommonLedgerProtocolPacket(TYPE_RESPONSE, requestId, writer.getBuffer())
 }
 
 function deserializeResponse (buffer) {
-  const { type, requestId, contents } = readEnvelope(buffer)
-  const reader = new Reader(contents)
+  const { type, requestId, data } = readCommonLedgerProtocolPacket(buffer)
+  const reader = new Reader(data)
 
   const protocolData = readProtocolData(reader)
   return { requestId, protocolData }
 }
 
-function serializeError ({ ilp }, requestId, protocolData) {
+function serializeError (error, requestId, protocolData) {
   const writer = new Writer()
-  const packet = Buffer.from(ilp, 'base64')
-
-  writer.write(packet)
+  const ilpPacket = serializeIlpError(error)
+  writer.write(ilpPacket)
   writeProtocolData(writer, protocolData)
 
-  return writeEnvelope(TYPE_ERROR, requestId, writer.getBuffer())
+  return writeCommonLedgerProtocolPacket(TYPE_ERROR, requestId, writer.getBuffer())
 }
 
 function deserializeError (buffer) {
-  const { type, requestId, contents } = readEnvelope(buffer)
-  const reader = new Reader(contents)
-
-  const ilp = readIlpPacket(reader)
+  const { type, requestId, data } = readCommonLedgerProtocolPacket(buffer)
+  const reader = new Reader(data)
+  const rejectionReason = readIlpError(reader)
   const protocolData = readProtocolData(reader)
-  return { requestId, ilp, protocolData }
+  return { requestId, rejectionReason, protocolData }
 }
 
-function serializePrepare ({ id, amount, executionCondition, expiresAt }, requestId, protocolData) {
-  const idBuffer = Buffer.from(id.replace(/\-/g, ''), 'hex')
+function serializePrepare ({ transferId, amount, executionCondition, expiresAt }, requestId, protocolData) {
```

Comment:
I tried to keep these field names as close to LPI fields as possible, which is why transferId is id. I don't think we use the field name `transferId` anywhere else in the codebase.